### PR TITLE
 NEW: Retrieve vat details from the Greek Ministry of Finance GSIS SOAP web service and autocomplete third party fields

### PR DESCRIPTION
**Pull request by sonikf**
NEW: Retrieve vat details from the Greek Ministry of Finance GSIS SOAP web service and autocomplete third party fields 

Re up messed up previous pr

### CODE REVIEW

File: htdocs/societe/card.php [hunk 1]
```diff
@@ -63,6 +65,12 @@
 	require_once DOL_DOCUMENT_ROOT.'/eventorganization/class/conferenceorboothattendee.class.php';
 }
 
+if ($mysoc->country_code == 'GR') {
+	$u = $conf->global->AADE_WEBSERVICE_USER;
```
Comment (eldy) [inline htdocs/societe/card.php:69]: `$conf->global->xxx `
is now deprecated. Use instead
`getDolGlobalString('xxx');`

File: htdocs/societe/card.php [hunk 2]
```diff
@@ -3339,3 +3359,46 @@ function init_check_no_email(input) {
...
+
+<script>
+// Function to retrieve VAT details from the Greek Ministry of Finance GSIS SOAP web service
+function GRVAT(a, u, p, myafm) {
+  var afm = a.replace(/\D/g, ""); // Remove non-digit characters from 'a'
+
+  $.ajax({
+	type: "GET",
+	url: "/societe/checkvat/checkVatGr.php",
+	data: { u: u, p: p, myafm: myafm, afm: afm }, // Set request parameters
```
Comment (eldy) [inline htdocs/societe/card.php:3373]: There is no need to propagate user and pass as parameter because there are global constant.

File: htdocs/societe/checkvat/checkVatGr.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ *  \file       htdocs/societe/checkvat/checkVatGr.php
+ *  \ingroup    societe
+ *  \brief      Request VAT details from the Greek Ministry of Finance GSIS SOAP web service
+ */
+
```
Comment (eldy) [inline htdocs/societe/checkvat/checkVatGr.php:25]: All php pages of dolibarr must first include the main.inc.php
If the page can be called without being authenticated, before the main you must include define('NOLOGIN', 1);

File: htdocs/societe/checkvat/checkVatGr.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+$username = (isset($_REQUEST['u']) ? $_REQUEST['u'] : ''); // Get username from request
```
Comment (eldy) [inline htdocs/societe/checkvat/checkVatGr.php:26]: param user and pass should be retreived with getDolGlobalString('MYDATA_AADE_KEY') or any other constant into setup.

Comment (sonikf) [reply htdocs/societe/checkvat/checkVatGr.php]: @eldy all done thank you! 

A quick search revealed 1.1k instances of $conf->global->xxx in Dolibarr code and also in translations strings.
Would you like this fixed ?

File: htdocs/societe/checkvat/checkVatGr.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+$username = getDolGlobalString('AADE_WEBSERVICE_USER'); // Get username from request
+$password = getDolGlobalString('AADE_WEBSERVICE_KEY'); // Get password from request
+$myafm = getDolGlobalString('MAIN_INFO_TVAINTRA'); // Get Vat from request
+$afm = (isset($_REQUEST['afm']) ? $_REQUEST['afm'] : ''); // Get client Vat from request
+
+// Make call to check VAT for Greek client
+$result = checkVATGR($username, $password, $myafm, $afm);
+
+echo json_encode($result); // Encode the result as JSON and output
```
Comment (eldy) [inline htdocs/societe/checkvat/checkVatGr.php:36]: you must add before the echo json_encode

top_httphead('application/json');

File: htdocs/societe/checkvat/checkVatGr.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ *  \ingroup    societe
+ *  \brief      Request VAT details from the Greek Ministry of Finance GSIS SOAP web service
+ */
+
+require "../../main.inc.php";
+
+$username = getDolGlobalString('AADE_WEBSERVICE_USER'); // Get username from request
+$password = getDolGlobalString('AADE_WEBSERVICE_KEY'); // Get password from request
+$myafm = getDolGlobalString('MAIN_INFO_TVAINTRA'); // Get Vat from request
+$afm = (isset($_REQUEST['afm']) ? $_REQUEST['afm'] : ''); // Get client Vat from request
```
Comment (eldy) [inline htdocs/societe/checkvat/checkVatGr.php:31]: Do not use $_REQUEST
Use instead
$afm = GETPOST('afm');

File: htdocs/societe/card.php [hunk 7]
```diff
@@ -1728,7 +1736,11 @@ function formatCustomerSelection (selection) {
 				print "\n";
 				print '<script type="text/javascript">';
 				print "function CheckVAT(a) {\n";
-				print "newpopup('".DOL_URL_ROOT."/societe/checkvat/checkVatPopup.php?vatNumber='+a, '".dol_escape_js($langs->trans("VATIntraCheckableOnEUSite"))."', ".$widthpopup.", ".$heightpopup.");\n";
+				if ($mysoc->country_code == 'GR' && $object->country_code == 'GR') {
```
Comment (eldy) [inline htdocs/societe/card.php:1739]: Can you add a test on   !empty($u)
so the GRVAT is used only for GR AND if setup of web service is defined.
Itfallback on common webservice for all european countries it not.

File: htdocs/societe/checkvat/checkVatGr.php (hunk 6 above)
Comment (sonikf) [reply htdocs/societe/checkvat/checkVatGr.php:31]: fixed

File: htdocs/societe/card.php (hunk 7 above)
Comment (sonikf) [reply htdocs/societe/card.php:1739]: test added

File: htdocs/societe/checkvat/checkVatGr.php (hunk 5 above)
Comment (sonikf) [reply htdocs/societe/checkvat/checkVatGr.php:37]: fixed

File: htdocs/societe/card.php [hunk 8]
```diff
@@ -3339,3 +3359,46 @@ function init_check_no_email(input) {
...
+
+<script>
+// Function to retrieve VAT details from the Greek Ministry of Finance GSIS SOAP web service
+function GRVAT(a, u, p, myafm) {
+  var afm = a.replace(/\D/g, ""); // Remove non-digit characters from 'a'
+
+  $.ajax({
+	type: "GET",
+	url: "/societe/checkvat/checkVatGr.php",
+	data: { u, p, myafm, afm }, // Set request parameters
```
Comment (eldy) [inline htdocs/societe/card.php:3373]: I think only the param afm must be oassed as parameter the the json ajax service. Now security data as retreived directly by the service and are no more exposed as parameter. Can you check with only afm and confirm my thinking ?

Comment (sonikf) [reply htdocs/societe/card.php]: Yes it works, fixed thank you.
